_posts/2021-10-01-gs_00004_basic_shell_script.md

```diff
@@ -0,0 +1,318 @@
+shell.
+
+# Your first Bash Script
+
+Every morning the same procedure. You go to work, get your first coffee and then
+open all required software for your job: Email client, meeting software,
+browser, music app, and so on. Probably it might be different depending on your
+daily work schedule. Wouldn't it be cool that at least the software opening part
+is done automatically? Let's take it as an example for the first script and open
+a browser. You already know how to do it with a terminal command, don't you? To
+write your first script, open a new file:
+
+    $ nano open_firefox.sh
+
+<div class='note-normal' markdown="1">
+_Note: '.sh' is the typical file extension for shell scripts_.
+</div>
+
+Enter the following text:
+
+    #!/bin/bash
+
+    # Print something to the terminal
+    echo "Greetings from Bits&Volts (bitsandvolts.org)"
+
+    # Open firefox
+    firefox &
+
+Press CTRL-X to close and 'y' (yes) to save the file. Before we run it, let's
+talk about it. The first line starting with the '#!' tells by what the script
+shall be interpreted. In our case, we want to use **Bash**. What does the 'bin'
+mean? The **bin** folder is a common folder where software programs such as the
+bash shell are installed. So, '/bin/bash/' is nothing but the absolute path of
+the shell program. The third line starting with '#' is just a comment that is
+not interpreted. If you write complex scripts, it is helpful to put some
+commands, so that someone else can better understand it. 'echo' is a command
+that we did not cover yet. It simply outputs some text. Then, we have another
+comment and the command to open firefox you already know. Note that we have to
+use the '&' here, as explained in our last post. Try to run the script as you
+would run a program:
+
+    $ open_firefox.sh
+
+This will not work as the command cannot be found. To understand why we have to
+look a bit deeper into what happens if you run a program. Clearly, the program
+has to be stored somewhere. But how can your operating system know where? The
+answer is: It can't! It only can look into all your files and folders and search
+for them. However, you can imagine that this can take a while, especially for
+large systems. This is why there are predefined folders where the programs have
+to be. As we mentioned earlier, that '/bin' is, per default, such a folder. You
+have three options:
+
+1. Move the script to a folder where it can be found. We do not want to do it
+   here as it is not always practical in real-life situations.
+2. Add your current folder to the so-called **PATH** variable. We explain in a
+   later post how to do it. It is also not always a good solution. If you add
+   too many folders it will take long to execute a command.
+3. Run the program by its absolute path. This is always an easy solution.
+   ```
+   $ ./open_firefox.sh
+   ```
+
+Ups... Still does not work. But how can it be that the permission is denied? You
```

Comment:
Maybe copy the error text here, otherwise the reader might get confused when he reads "permission denied" (not all readers might be running the commands themselves)

Reply:
recheck